NBitcoin/Protocol/Payloads/VersionPayload.cs

```diff
@@ -1,235 +1,238 @@
-#else
-				var version = typeof(VersionPayload).GetTypeInfo().Assembly.GetName().Version;
-#endif
-				_NUserAgent = "/NBitcoin:" + version.Major + "." + version.MajorRevision + "." + version.Build + "/";
-#endif
-
-			}
-			return _NUserAgent;
-		}
-		uint version;
-
-		public uint Version
-		{
-			get
-			{
-				return version;
-			}
-			set
-			{
-				version = value;
-			}
-		}
-		ulong services;
-
-		public NodeServices Services
-		{
-			get
-			{
-				return (NodeServices)services;
-			}
-			set
-			{
-				services = (ulong)value;
-			}
-		}
-
-		long timestamp;
-
-		public DateTimeOffset Timestamp
-		{
-			get
-			{
-				return Utils.UnixTimeToDateTime((uint)timestamp);
-			}
-			set
-			{
-				timestamp = Utils.DateTimeToUnixTime(value);
-			}
-		}
-
-		NetworkAddress addr_recv = new NetworkAddress();
-		public IPEndPoint AddressReceiver
-		{
-			get
-			{
-				return addr_recv.Endpoint;
-			}
-			set
-			{
-				addr_recv.Endpoint = value;
-			}
-		}
-		NetworkAddress addr_from = new NetworkAddress();
-		public IPEndPoint AddressFrom
-		{
-			get
-			{
-				return addr_from.Endpoint;
-			}
-			set
-			{
-				addr_from.Endpoint = value;
-			}
-		}
-
-		ulong nonce;
-		public ulong Nonce
-		{
-			get
-			{
-				return nonce;
-			}
-			set
-			{
-				nonce = value;
-			}
-		}
-		int start_height;
-
-		public int StartHeight
-		{
-			get
-			{
-				return start_height;
-			}
-			set
-			{
-				start_height = value;
-			}
-		}
-
-		bool relay;
-		public bool Relay
-		{
-			get
-			{
-				return relay;
-			}
-			set
-			{
-				relay = value;
-			}
-		}
-
-		VarString user_agent;
-		public string UserAgent
-		{
-			get
-			{
-				return Encoders.ASCII.EncodeData(user_agent.GetString());
-			}
-			set
-			{
-				user_agent = new VarString(Encoders.ASCII.DecodeData(value));
-			}
-		}
-
-		#region IBitcoinSerializable Members
-
-		public override void ReadWriteCore(BitcoinStream stream)
-		{
-			stream.ReadWrite(ref version);
-			using (stream.ProtocolVersionScope(version))
-			{
-				stream.ReadWrite(ref services);
-				stream.ReadWrite(ref timestamp);
-				using (stream.SerializationTypeScope(SerializationType.Hash)) //No time field in version message
-				{
-					stream.ReadWrite(ref addr_recv);
-				}
-				if (version >= 106)
-				{
-					using (stream.SerializationTypeScope(SerializationType.Hash)) //No time field in version message
-					{
-						stream.ReadWrite(ref addr_from);
-					}
-					stream.ReadWrite(ref nonce);
-					stream.ReadWrite(ref user_agent);
-					if (!stream.ProtocolCapabilities.SupportUserAgent)
-						if (user_agent.Length != 0)
-							throw new FormatException("Should not find user agent for current version " + version);
-					stream.ReadWrite(ref start_height);
-					if (version >= 70001)
-						stream.ReadWrite(ref relay);
-				}
-			}
-		}
-
-		#endregion
-
-
-		public override string ToString()
-		{
-			return Version.ToString();
-		}
-	}
-}
+#if !NOSOCKET
```

Comment:
It looks like that your editor changed formatting of this file - tabs vs spaces?

Reply:
Oh F***! i don't know when that happened. I need to fix it. Thank you.